mlos_core/mlos_core/optimizers/optimizer.py

```diff
@@ -27,6 +27,7 @@ class BaseOptimizer(metaclass=ABCMeta):
     def __init__(self, *,
                  parameter_space: ConfigSpace.ConfigurationSpace,
                  optimization_targets: List[str],
+                 objective_weights: Optional[List[float]] = None,
```

Comment:
I was thinking of it but decided to keep it simple and stick closer to SMAC3 API. We are yet to implement the objective weights in mlos_bench anyway, and we'll have to decide on the API and JSON representation. That is, instead of 
```javascript
"optimization_targets": {
    "score": "max",
    "other_score": "min"
}
```
we'll have to come up with something like 
```javascript
"optimization_targets": {
    "score": {
        "direction": "max",
        "weight": 0.7
    },
    "other_score": {
        "direction": "min",
        "weight": 0.3
    }
}
```
and then maybe consider some pluggable optimization policies that might have other parameters instead of/in addition to weights...

so I would rather keep `objective_weights` as a separate parameter for now, just to tick all boxes for SMAC and enable multi-target optimization in FLAML, + have unit tests for multi-target optimization for all optimizers that we support. WDYT?

Reply:
The whole point of mlos_core is to try and abstract away the APIs of other libraries that we wrap, so to me the argument of "that's what SMAC does so we should follow it too" doesn't really hold water for me.

Array arguments that must match in alignment and size is just such a C-ism.  And don't get me wrong, I love me some low-level C, but I feel like we can do better.

This change seems kinda simple to me actually.

Internally you could even split the passed `Dict[str, float]` back out again if you wanted to so that in the object we kept the aligned lists that you have now.

But then it's on us to manage and mask, not the user.

But you're right, on the mlos_bench side, we need to change the json schema too and I think we already have that same problem there with the way we did weighted distributions for tunables, so maybe it doesn't matter.

`</cent></cent>`

Reply:
I'd say, let's merge it as is just to get it out of the way - we have a lot of other issues to deal with this Summer, including the security stuff and workload identification, and no one will be using multi-target optimization in earnest any time soon

Reply:
Works for me.  Approved.  Merged.